docs/src/features.md

```diff
@@ -7,26 +7,80 @@
 
-This section explains the main features in the optimization model inside the model based on these definitions.
+The model guarantees a balance of energy for the various types of assets while considering the flow limits. The [`mathematical formulation`](@ref math-formulation) defines the flow variable ($v^{flow}_{f,rp,k}$) as the instantaneous value (e.g., power in MW) for each flow $f$ between two assets, representative period $rp$, and time step $k$. The time step $k$ can represent a single time step (e.g., 1, 2, 3...) or a range of time steps (e.g., 1:2, meaning that the variable represents the value of both time steps 1 and 2). For more examples and details on this topic, refer to the section on [flexible time resolution](@ref flex-time-res).
 
-## [Flexible Connexion of Energy Assets](@id flex-asset-connexion)
+The following sections explain the main features in the optimization model inside the model based on all these concepts and definitions.
 
-The representation of the energy system in *TulipaEnergyModel.jl* is based on [Graph Theory](https://www.britannica.com/topic/graph-theory), which deals with connection amongst vertices by edges. This representation provides a more flexible framework to model *energy assets* in the system as *vertices* and *flows* between energy assets as *edges*. In addition, it poses some advantages from the modelling perspective. For instance, connecting assets directly to each other without having a node between them allows for a reduction of the number of variables and constraints to represent different configurations (e.g., co-location of energy storage and renewable units to model a hybrid operation).
+## [Flexible Connection of Energy Assets](@id flex-asset-connection)
+
+The representation of the energy system in *TulipaEnergyModel.jl* is based on [Graph Theory](https://www.britannica.com/topic/graph-theory), which deals with connection amongst vertices by edges. This representation provides a more flexible framework to model *energy assets* in the system as *vertices* and *flows* between energy assets as *edges*. In addition, it poses some advantages from the modeling perspective. For instance, connecting assets directly to each other without having a node between them allows for a reduction of the number of variables and constraints to represent different configurations (e.g., co-location of energy storage and renewable units to model a hybrid operation).
+
+Consider the following example to demonstrate the benefits of this approach. In the classic connection approach, the nodes play a crucial role in modeling. For example, every asset needs to be connected to a node that has balance constraints attached to it. Therefore, to include the co-location of a storage asset and a renewable asset, we must introduce a node between these assets and the external power grid, as shown in the following figure:
```

Comment:
Regarding your questions:
1. co-location refers to location and operation (technically connected). Another example, apart from the one in the documentation, is a battery + solar configuration, where both are in the exact location, and the battery goal is to reduce the solar PV curtailment, so it only charges from the solar PV (they can't charge from the grid). I think there is no standard name for this configuration (hybrid operation is also sometimes used, but maybe co-location is not a good name; we can come up with a better name if you have any other suggestions).
2. If they are both connected to the balance node directly, then you can't model the fact that the storage can't charge from the external grid. So, from the optimization point of view, it would be the same as if they were individual assets (and you wouldn't get that the storage only charges from the solar PV).
3. Both are balancing nodes (the thing is that we don't have "nodes" in Tulipa); I'm not good at giving names. I want a short name in one word. I am also trying to change the rest of the written section as little as possible. Anyhow, maybe I need to change the names. Any ideas?

Another example of co-location (or hybrid operation) is when you want to model green-hydrogen production from an electrolyzer + renewable configuration. You can only guarantee that the hydrogen production is green if the electrolyzer produces hydrogen directly from the renewable asset. Since renewable assets might also produce energy for the power grid, you need (as in the previous case) to avoid the electrolyzer consuming (charging) electricity from the grid.

We may need to add more details to the section. But we need to find a balance because I also don't want to explain too many concepts that are not the focus of the section. I'm open to suggestions.